File: src/ProjectMapData/RouteMap.ts
import { PageMap } from "./PageMap"

export interface RouteMap
{
    RelativePathSegments: string[]  // TODO: now calculated in ts, maybe optmize - calc RelativePath in advance
    Name: string
    ChildRoutes: RouteMap[]
    Pages: PageMap[]
    //PathSegments: string[]
    //ResourceFiles: string[]
}

File: ProjectMapLanguageServer/ProjectMapSg/ContractModels/RouteMap.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectMapSg.ContractModels
{
    public class RouteMap
    {
        public RouteMap(string name)
        {
            Name = name;
        }

        public string Name { get; }

        public List<RouteMap> ChildRoutes{ get; set;} = new List<RouteMap>();

        public string Debug {get; set;}

        public List<PageMap> Pages { get; set; } = new List<PageMap>();
    }
}
